File: katas/7 kyu/beginner-series-number-3-sum-of-numbers/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def get_sum(a,b):
  return sum(range(min(a,b),max(a,b)+1))
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def get_sum(a, b):
    return (a + b) * (abs(a - b) + 1) // 2
# ----------------------------------- TEST ----------------------------------- #
@test.describe('Sum of numbers')
def desc1():
    @test.it('Sample tests')
    def it1():
        test.assert_equals(get_sum(0,1),1)
        test.assert_equals(get_sum(0,-1),-1)

File: katas/7 kyu/beginner-series-number-3-sum-of-numbers/solution.js
// beginner-series-number-3-sum-of-numbers
/* ------------------------------- MY SOLUTION ------------------------------ */
function getSum( a,b )
{
  if (a == b) return a
  if (b < a) arr = [...Array(a-b+1).keys()].map(x => a-x)
  else arr = [...Array(b-a+1).keys()].map(x => a+x)
  return arr.reduce((a, b) => a+b , 0)
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
const GetSum = (a, b) => {
  let min = Math.min(a, b),
      max = Math.max(a, b);
  return (max - min + 1) * (min + max) / 2;
}
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Basic tests", () => {
  it("Testing for fixed tests", () => {         
    assert.strictEqual(getSum(0,-1),-1);
    assert.strictEqual(getSum(0,1),1);
  })
});